diff --git a/warband/module-system/Module_system 1.171/module_game_menus.py b/warband/module-system/Module_system 1.171/module_game_menus.py
index 040ec82..12ef859 100644
--- a/warband/module-system/Module_system 1.171/module_game_menus.py	
+++ b/warband/module-system/Module_system 1.171/module_game_menus.py	
@@ -7531,6 +7531,39 @@ game_menus = [
            (assign, "$g_next_menu", "mnu_village"),
            (jump_to_menu, "mnu_center_manage"),
         ]),
+
+      ("village_elder_talk",[(neg|party_slot_eq, "$current_town", slot_village_state, svs_looted),
+                        (neg|party_slot_eq, "$current_town", slot_village_state, svs_being_raided),
+                        (neg|party_slot_ge, "$current_town", slot_village_infested_by_bandits, 1),]
+      ,"Speak with the Village Elder.",
+      [
+        (try_begin),
+          (call_script, "script_cf_enter_center_location_bandit_check"),
+        (else_try),
+          (party_get_slot, ":village_scene", "$current_town", slot_castle_exterior),
+          (modify_visitors_at_site,":village_scene"),
+          (reset_visitors),
+          (party_get_slot, ":village_elder_troop", "$current_town",slot_town_elder),
+          (set_visitor, 11, ":village_elder_troop"),
+
+          (call_script, "script_init_town_walkers"),
+
+          (try_begin),
+            (check_quest_active, "qst_hunt_down_fugitive"),
+            (neg|is_currently_night),
+            (quest_slot_eq, "qst_hunt_down_fugitive", slot_quest_target_center, "$current_town"),
+            (neg|check_quest_succeeded, "qst_hunt_down_fugitive"),
+            (neg|check_quest_failed, "qst_hunt_down_fugitive"),
+            (set_visitor, 45, "trp_fugitive"),
+          (try_end),
+
+          (set_jump_mission,"mt_village_center"),
+          (jump_to_scene,":village_scene"),
+
+          (change_screen_map_conversation, ":village_elder_troop"),
+        (try_end),
+        ],"Door to the village center."),
+
       ("recruit_volunteers",
       [
         (call_script, "script_cf_village_recruit_volunteers_cond"),
