diff --git a/warband/module-system/Module_system 1.171/module_game_menus.py b/warband/module-system/Module_system 1.171/module_game_menus.py
index 040ec82..8bd799d 100644
--- a/warband/module-system/Module_system 1.171/module_game_menus.py	
+++ b/warband/module-system/Module_system 1.171/module_game_menus.py	
@@ -4977,12 +4977,6 @@ game_menus = [
       ],
     [
       ("pre_join_help_attackers",[
-          (store_faction_of_party, ":attacker_faction", "$g_encountered_party_2"),
-          (store_relation, ":attacker_relation", ":attacker_faction", "fac_player_supporters_faction"),
-          (store_faction_of_party, ":defender_faction", "$g_encountered_party"),
-          (store_relation, ":defender_relation", ":defender_faction", "fac_player_supporters_faction"),
-          (ge, ":attacker_relation", 0),
-          (lt, ":defender_relation", 0),
           ],
           "Move in to help the {s2}.",[
               (select_enemy,0),
@@ -4990,12 +4984,6 @@ game_menus = [
               (assign,"$g_ally_party","$g_encountered_party_2"),
               (jump_to_menu,"mnu_join_battle")]),
       ("pre_join_help_defenders",[
-          (store_faction_of_party, ":attacker_faction", "$g_encountered_party_2"),
-          (store_relation, ":attacker_relation", ":attacker_faction", "fac_player_supporters_faction"),
-          (store_faction_of_party, ":defender_faction", "$g_encountered_party"),
-          (store_relation, ":defender_relation", ":defender_faction", "fac_player_supporters_faction"),
-          (ge, ":defender_relation", 0),
-          (lt, ":attacker_relation", 0),
           ],
           "Rush to the aid of the {s1}.",[
               (select_enemy,1),
