diff --git a/warband/module-system/Module_system 1.171/module_game_menus.py b/warband/module-system/Module_system 1.171/module_game_menus.py
index 040ec82..171ede3 100644
--- a/warband/module-system/Module_system 1.171/module_game_menus.py	
+++ b/warband/module-system/Module_system 1.171/module_game_menus.py	
@@ -7770,7 +7770,13 @@ game_menus = [
       ],
       "I don't have enough money...",
       [
-        (jump_to_menu,"mnu_village"),
+        (try_begin),
+          (this_or_next|party_slot_eq, "$current_town", slot_party_type, spt_castle),
+          (party_slot_eq, "$current_town", slot_party_type, spt_town),
+          (jump_to_menu, "mnu_town"),
+        (else_try),
+          (jump_to_menu, "mnu_village"),
+        (try_end),
       ]),
        
       ("continue",
@@ -7781,7 +7787,13 @@ game_menus = [
       "Continue...",
       [
         (party_set_slot, "$current_town", slot_center_volunteer_troop_amount, -1),
-        (jump_to_menu,"mnu_village"),
+        (try_begin),
+          (this_or_next|party_slot_eq, "$current_town", slot_party_type, spt_castle),
+          (party_slot_eq, "$current_town", slot_party_type, spt_town),
+          (jump_to_menu, "mnu_town"),
+        (else_try),
+          (jump_to_menu, "mnu_village"),
+        (try_end),
       ]),
        
       ("recruit_them",
@@ -7793,7 +7805,13 @@ game_menus = [
       [
         (call_script, "script_village_recruit_volunteers_recruit"),
                         
-        (jump_to_menu,"mnu_village"),
+        (try_begin),
+          (this_or_next|party_slot_eq, "$current_town", slot_party_type, spt_castle),
+          (party_slot_eq, "$current_town", slot_party_type, spt_town),
+          (jump_to_menu, "mnu_town"),
+        (else_try),
+          (jump_to_menu, "mnu_village"),
+        (try_end),
       ]),
       
       ("forget_it",
@@ -7803,7 +7821,13 @@ game_menus = [
       ],
       "Forget it.",
       [
-        (jump_to_menu,"mnu_village"),
+        (try_begin),
+          (this_or_next|party_slot_eq, "$current_town", slot_party_type, spt_castle),
+          (party_slot_eq, "$current_town", slot_party_type, spt_town),
+          (jump_to_menu, "mnu_town"),
+        (else_try),
+          (jump_to_menu, "mnu_village"),
+        (try_end),
       ]),
     ],
   ),
@@ -9582,6 +9606,19 @@ game_menus = [
         (jump_to_menu, "mnu_lady_visit"),
         ], "Door to the garden."),										
 		
+      ("recruit_volunteers",
+      [
+        (call_script, "script_cf_town_castle_recruit_volunteers_cond"),
+       ]
+       ,"Recruit Volunteers.",
+       [
+         (try_begin),
+           (call_script, "script_cf_enter_center_location_bandit_check"),
+         (else_try),
+           (jump_to_menu, "mnu_recruit_volunteers"),
+         (try_end),
+        ]),
+
       ("trade_with_merchants",
        [
            (party_slot_eq,"$current_town",slot_party_type, spt_town)
diff --git a/warband/module-system/Module_system 1.171/module_scripts.py b/warband/module-system/Module_system 1.171/module_scripts.py
index e756cc4..9744213 100644
--- a/warband/module-system/Module_system 1.171/module_scripts.py	
+++ b/warband/module-system/Module_system 1.171/module_scripts.py	
@@ -33944,6 +33944,25 @@ scripts = [
 	 
      ]),
 
+  #script_cf_town_castle_recruit_volunteers_cond
+  # INPUT: none
+  # OUTPUT: none
+  ("cf_town_castle_recruit_volunteers_cond",
+    [(store_faction_of_party, ":town_faction", "$current_town"),
+     (party_get_slot, ":center_relation", "$current_town", slot_center_player_relation),
+     (store_relation, ":town_faction_relation", ":town_faction", "fac_player_faction"),
+     (ge, ":center_relation", 0),
+     (this_or_next|ge, ":center_relation", 5),
+     (this_or_next|eq, ":town_faction", "$players_kingdom"),
+     (this_or_next|ge, ":town_faction_relation", 0),
+     (this_or_next|eq, ":town_faction", "$supported_pretender_old_faction"),
+     (             eq, "$players_kingdom", 0),
+     (party_slot_ge, "$current_town", slot_center_volunteer_troop_amount, 0),
+     (party_slot_ge, "$current_town", slot_center_volunteer_troop_type, 1),
+     (party_get_free_companions_capacity, ":free_capacity", "p_main_party"),
+     (ge, ":free_capacity", 1),
+     ]),
+
   #script_village_recruit_volunteers_recruit
   # INPUT: none
   # OUTPUT: none
diff --git a/warband/module-system/Module_system 1.171/module_simple_triggers.py b/warband/module-system/Module_system 1.171/module_simple_triggers.py
index 1a63807..cddcf2e 100644
--- a/warband/module-system/Module_system 1.171/module_simple_triggers.py	
+++ b/warband/module-system/Module_system 1.171/module_simple_triggers.py	
@@ -2525,7 +2525,7 @@ simple_triggers = [
      (call_script, "script_update_tavern_minstrels"),
      (call_script, "script_update_booksellers"),
      (call_script, "script_update_villages_infested_by_bandits"),
-     (try_for_range, ":village_no", villages_begin, villages_end),
+     (try_for_range, ":village_no", centers_begin, centers_end),
        (call_script, "script_update_volunteer_troops_in_village", ":village_no"),
        (call_script, "script_update_npc_volunteer_troops_in_village", ":village_no"),
      (try_end),
